### PR TITLE
Add SourceLink

### PR DESCRIPTION
**Pull request by fgreinacher**
This PR adds [SourceLink](https://github.com/dotnet/sourcelink) support to our assemblies/PDBs.

I verified it using `sourcelink test` as per [the documentation](https://github.com/ctaggart/sourcelink/#test) and everything looks good.

### CODE REVIEW

**Comment by fgreinacher** [comment]
Any objections on merging this?

**Comment by jpreese** [comment]
This actually seems really cool. This has always been a big issue at my place of work (the need to debug packages), and we've always had to do some pretty big heroics to get that functionality.

For this, all I'd want to do is test it. Intentionally break something in the library, create a local package, reference said package, and see what the debugging story is like.

**Comment by jpreese** [comment]
Hm.. I'm no longer able to create a NuGet package when I checkout this branch locally (dotnet pack). The error is quite lengthy, but here's a snippet.

C:\Users\jpree\.nuget\packages\microsoft.build.tasks.git\1.0.0-beta-63102-01\build\Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: The "Microsoft.Build.Tasks.Git.LocateRepository" task failed unexpectedly.\r [C:\Users\jpree\Desktop\System.IO.Abstractions\System.IO.Abstractions\System.IO.Abstractions.csproj]
C:\Users\jpree\.nuget\packages\microsoft.build.tasks.git\1.0.0-beta-63102-01\build\Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018: System.TypeInitializationException: The type initializer for 'LibGit2Sharp.Core.NativeMethods' threw an exception. ---> System.IO.FileNotFoundException: Could not load file or assembly 'Microsoft.DotNet.PlatformAbstractions, Version=2.1.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. The system cannot find the file specified.\r [C:\Users\jpree\Desktop\System.IO.Abstractions\System.IO.Abstractions\System.IO.Abstractions.csproj]
C:\Users\jpree\.nuget\packages\microsoft.build.tasks.git\1.0.0-beta-63102-01\build\Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.RuntimeIdMap.GetNativeLibraryDirectoryName()\r [C:\Users\jpree\Desktop\System.IO.Abstractions\System.IO.Abstractions\System.IO.Abstractions.csproj]
C:\Users\jpree\.nuget\packages\microsoft.build.tasks.git\1.0.0-beta-63102-01\build\Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.GitLoaderContext.GetNativeLibraryDirectory() in /_/src/Microsoft.Build.Tasks.Git/GitLoaderContext.cs:line 55\r [C:\Users\jpree\Desktop\System.IO.Abstractions\System.IO.Abstractions\System.IO.Abstractions.csproj]
C:\Users\jpree\.nuget\packages\microsoft.build.tasks.git\1.0.0-beta-63102-01\build\Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at Microsoft.Build.Tasks.Git.GitLoaderContext.LoadUnmanagedDll(String unmanagedDllName) in /_/src/Microsoft.Build.Tasks.Git/GitLoaderContext.cs:line 30\r [C:\Users\jpree\Desktop\System.IO.Abstractions\System.IO.Abstractions\System.IO.Abstractions.csproj]
C:\Users\jpree\.nuget\packages\microsoft.build.tasks.git\1.0.0-beta-63102-01\build\Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at System.Runtime.Loader.AssemblyLoadContext.ResolveUnmanagedDll(String unmanagedDllName, IntPtr gchManagedAssemblyLoadContext)\r [C:\Users\jpree\Desktop\System.IO.Abstractions\System.IO.Abstractions\System.IO.Abstractions.csproj]
C:\Users\jpree\.nuget\packages\microsoft.build.tasks.git\1.0.0-beta-63102-01\build\Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.git_libgit2_init()\r [C:\Users\jpree\Desktop\System.IO.Abstractions\System.IO.Abstractions\System.IO.Abstractions.csproj]
C:\Users\jpree\.nuget\packages\microsoft.build.tasks.git\1.0.0-beta-63102-01\build\Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods.InitializeNativeLibrary()\r [C:\Users\jpree\Desktop\System.IO.Abstractions\System.IO.Abstractions\System.IO.Abstractions.csproj]
C:\Users\jpree\.nuget\packages\microsoft.build.tasks.git\1.0.0-beta-63102-01\build\Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    at LibGit2Sharp.Core.NativeMethods..cctor()\r [C:\Users\jpree\Desktop\System.IO.Abstractions\System.IO.Abstractions\System.IO.Abstractions.csproj]
C:\Users\jpree\.nuget\packages\microsoft.build.tasks.git\1.0.0-beta-63102-01\build\Microsoft.Build.Tasks.Git.targets(20,5): error MSB4018:    --- End of inner exception stack trace ---\r [C:\Users\jpree\Desktop\System.IO.Abstractions\System.IO.Abstractions\System.IO.Abstractions.csproj]

**Comment by fgreinacher** [comment]
Do you have .NET Core SDK 2.1.300? This seems so be the minimum supported version as per https://github.com/dotnet/sourcelink/issues/107

File: System.IO.Abstractions/System.IO.Abstractions.csproj [hunk 1]
```diff
@@ -47,4 +53,8 @@
         <Reference Include="System" />
         <Reference Include="Microsoft.CSharp" />
     </ItemGroup>
+
+    <ItemGroup>
+        <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-beta-63102-01" PrivateAssets="All"/>
```
Comment (herebebeasties) [inline System.IO.Abstractions/System.IO.Abstractions.csproj]:
```suggestion
        <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-beta-63127-02" PrivateAssets="All"/>
```

File: TestingHelpers/TestingHelpers.csproj [hunk 2]
```diff
@@ -36,4 +42,8 @@
         <PackageReference Include="System.Runtime.Serialization.Formatters" Version="4.3.0" />
         <PackageReference Include="System.Security.AccessControl" Version="4.5.0" />
     </ItemGroup>
+    
+    <ItemGroup>
+        <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-beta-63102-01" PrivateAssets="All"/>
```
Comment (herebebeasties) [inline TestingHelpers/TestingHelpers.csproj]:
```suggestion
        <PackageReference Include="Microsoft.SourceLink.GitHub" Version="1.0.0-beta-63127-02" PrivateAssets="All"/>
```

**Pull request closed without merging**